src/api/app/models/concerns/staging_project.rb

```diff
@@ -142,35 +142,22 @@ def cache_problems
   end
 
   def state
-    return :empty unless staged_requests.exists?
-    return :unacceptable if unacceptable_state?
-    return :review if review_state?
-    general_state
-  end
-
-  def unacceptable_state?
-    untracked_requests.present? || staged_requests.obsolete.present?
-  end
-
-  def review_state?
-    general_state == :acceptable && missing_reviews.present?
-  end
-
-  def general_state
-    build_state == :acceptable ? check_state : build_state
+    return :empty if staged_requests.blank?
+    return :unacceptable if untracked_requests.present? || staged_requests.obsolete.exists?
+    build_or_check_state = build_state || check_state
```

Comment:
`build_state` always return a value, which means that `check_state` will never be executed

Reply:
It does not. Behind every return is an if (including the silent last one)

Reply:
Exactly, in case `build_state` is not 'building' or 'failed' it will return with nil. Therefore, check_status will be called.

Reply:
I read it wrong, sorry